### DISCUSSION TITLE
Need some help understanding interaction with games.

### DISCUSSION THREAD

**Discussion by RobertZenz** [yshui/picom#681] [Q&A] [answered]
I've been using picom as compton replacement for a few months now, and it is working great. But I need some help with understanding how it does interact with games.

Following scenario, Sawfish+picom on nVidia hardware with VSync enabled for the desktop, that works. Many games that I play I want to have VSync enabled for them, because tearing is annoying from time to time. But I've run into some issues with that. First, if the game starts normally (picom VSync and game VSync enabled), it stutters, which is kinda to be expected I guess. So I've created a little helper which allows me to toggle `_NET_WM_BYPASS_COMPOSITOR` on the game window (if the game does not set it), so that picom disables VSync on the desktop level, that also works. But it also kills VSync in the game completely, so even if I toggle VSync in the game, I always have tearing.

Is that to be expected? Is that because the games are running as "borderless window" instead of "fullscreen" or something like that? Or did I misunderstand how the bypass-flag is supposed to work?

**Comment by tryone144** [comment, marked as answer]
`_NET_WM_BYPASS_COMPOSITOR` stops picom from rendering the screen (so effectively disables VSync on picom's side). Your X server most likely renders without VSync.

Running your game in a "borderless fullscreen window" renders the game inside an X window which is then displayed by the X server (and, as far as I know, consequently no VSync unless your display driver supports it).

**Reply by RobertZenz** [reply]
I see, that means that all the VSync I've seen so far in games has been purely compton (which did not go away when the game started) until I switched to picom. Well, that would explain a few things, actually. So I need to teach my X server how to VSync properly, I see.

Thanks for the help.
